src/pages/creer-une-enigme/4.tsx

```diff
@@ -1,105 +1,195 @@
           <p className="text" style={{ fontSize: '1.1rem' }}>
-            Créez ici un <strong>indice</strong> pour faire deviner votre {enigmeType.titleStep2Short}. Vous pouvez ajouter du texte, une vidéo ou une
-            image à votre indice et vous pourrez le modifier à l’étape 4.
+            Voici la pré-visualisation de votre énigme.
+            {isEdit
+              ? " Vous pouvez la modifier à l'étape précédente, et enregistrer vos changements ici."
+              : ' Vous pouvez la modifier, et quand vous êtes prêts : publiez-la dans votre village-monde !'}
           </p>
-          <ContentEditor
-            content={activity.content.slice(indiceContentIndex, activity.content.length)}
-            updateContent={updateContent}
-            addContent={addIndiceContent}
-            deleteContent={deleteIndiceContent}
-            save={save}
-          />
-          <StepsButton prev="/creer-une-enigme/3" next={onNext} />
+          {isEdit ? (
+            <div style={{ width: '100%', display: 'flex', justifyContent: 'space-between', margin: '1rem 0' }}>
+              <Link href="/creer-une-enigme/4" passHref>
+                <Button component="a" color="secondary" variant="contained" href="/creer-une-enigme/4">
+                  {"Modifier à l'étape précédente"}
+                </Button>
+              </Link>
+              <Button variant="outlined" color="primary" onClick={onPublish}>
+                Enregistrer les changements
+              </Button>
+            </div>
+          ) : (
+            <div style={{ width: '100%', textAlign: 'right', margin: '1rem 0' }}>
+              {!isValid && (
+                <p>
+                  <b>Avant de publier votre présentation, il faut corriger les étapes incomplètes, marquées en orange.</b>
+                </p>
+              )}
+              <div style={{ width: '100%', textAlign: 'right', margin: '1rem 0' }}>
+                <Button variant="outlined" color="primary" onClick={onPublish} disabled={!isValid}>
+                  Publier
+                </Button>
+              </div>
+            </div>
+          )}
+
+          {!isEdit && activity.responseActivityId === null && <ActivityLink url={`/creer-une-enigme/1?edit=${activity.id}`} />}
+          {responseActivity !== null && (
+            <>
+              <span className="text text--small text--success">Énigme en réaction à {REACTIONS[responseActivity.type]}</span>
+              <div className="preview-block">
+                {!isEdit && (
+                  <EditButton
+                    onClick={() => {
+                      router.push(`/creer-une-enigme/1?edit=${activity.id}`);
+                    }}
+                    status={'success'}
+                    style={{ position: 'absolute', top: '0.5rem', right: '0.5rem' }}
+                  />
+                )}
+                <Activities activities={[responseActivity]} noButtons />
+              </div>
+            </>
+          )}
```

Comment:
**Nit:** Maybe I didn't catch those in your previous PRs, but this would be great to do in another PR anyway. Can you remove all stuff related to `responseActivity` in these activities, now that there is a specfic one for that (Reaction activity). So we don't need that here.

Reply:
Oki I will do it in a global PR for this one.